contracts/finality/src/contract.rs

```diff
@@ -135,6 +138,29 @@ pub fn execute(
             max_signatures_to_prune,
             max_pub_rand_values_to_prune,
         ),
+        ExecuteMsg::AddToAllowlist { fp_pubkey_hex } => {
+            if fp_pubkey_hex.is_empty() {
+                return Err(ContractError::EmptyFpBtcPubKey);
+            }
+            ADMIN.assert_admin(deps.as_ref(), &info.sender)?;
+            crate::state::config::add_finality_provider_to_allowlist(deps.storage, &fp_pubkey_hex)?;
+            Ok(Response::new()
+                .add_attribute("action", "add_to_allowlist")
+                .add_attribute("fp_pubkey_hex", fp_pubkey_hex))
+        }
+        ExecuteMsg::RemoveFromAllowlist { fp_pubkey_hex } => {
+            if fp_pubkey_hex.is_empty() {
+                return Err(ContractError::EmptyFpBtcPubKey);
+            }
+            ADMIN.assert_admin(deps.as_ref(), &info.sender)?;
```

Comment:
One more thing, do you think it's worth allowing both contract admin and Babylon governance account to do removal. If contract admin goes rogue, Babylon governance can protect itself by removing everything in the allowlist.

But on the other hand, Babylon governance can simply remove the BSN from registry, or remove the contract. So I'm not sure if this is necessary

Reply:
I would keep it like this, for simplicity. At some point, the contract admin can surrender control to governance.